homeassistant/components/tradfri.py

```diff
@@ -96,35 +105,33 @@ def async_setup(hass, config):
     conf = config.get(DOMAIN, {})
     host = conf.get(CONF_HOST)
     allow_tradfri_groups = conf.get(CONF_ALLOW_TRADFRI_GROUPS)
-    keys = yield from hass.async_add_job(_read_config, hass)
+    known_hosts = yield from hass.async_add_job(_read_config, hass)
 
     @asyncio.coroutine
     def gateway_discovered(service, info):
         """Run when a gateway is discovered."""
         host = info['host']
 
-        if host in keys:
-            yield from _setup_gateway(hass, config, host, keys[host]['key'],
-                                      allow_tradfri_groups)
+        if host in known_hosts:
+            yield from _setup_gateway(hass, config, host,
+                                        known_hosts[host]['identity'],
+                                        known_hosts[host]['token'],
+                                        allow_tradfri_groups)
         else:
             hass.async_add_job(request_configuration, hass, config, host)
+            return True
```

Comment:
Why add this? The if Branch doesn't do that , consistent return is important. Is it used?